File: app/src/main/java/com/example/museumapplication/utils/GpsCheckUtils.java
package com.example.museumapplication.utils;

import android.content.Context;
import android.location.LocationManager;
import android.os.Build;

import androidx.annotation.RequiresApi;

public class GpsCheckUtils {
    private GpsCheckUtils() {}

    /**
     * Is Gps Enabled
     *
     * @param context Context
     * @return true:Gps is enabled
     */
    @RequiresApi(api = Build.VERSION_CODES.P)
    public static boolean isGpsEnabled(Context context) {
        Object object = context.getSystemService(Context.LOCATION_SERVICE);
        if (!(object instanceof LocationManager)) {
            return false;
        }
        LocationManager locationManager = (LocationManager) object;
        return locationManager.isLocationEnabled();
    }
}

File: app/src/main/java/com/example/museumapplication/utils/virtualGuide/GpsCheckUtils.kt
package com.example.museumapplication.utils.virtualGuide

import android.content.Context
import android.location.LocationManager
import android.os.Build
import androidx.annotation.RequiresApi

object GpsCheckUtils {
    /**
     * Is Gps Enabled
     *
     * @param context Context
     * @return true:Gps is enabled
     */
    @RequiresApi(api = Build.VERSION_CODES.P)
    fun isGpsEnabled(context: Context): Boolean {
        val `object` = context.getSystemService(Context.LOCATION_SERVICE) as? LocationManager
                ?: return false
        return `object`.isLocationEnabled
    }
}